src/client/app/components/visual-group/CreateVisualGroupComponent.tsx

```diff
@@ -0,0 +1,851 @@
+
+
+import * as React from 'react';
+import * as d3 from 'd3';
+import { useEffect, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useAppSelector } from '../../redux/reduxHooks';
+import { GroupData } from '../../../../client/app/types/redux/groups'
+import { MeterData } from '../../../../client/app/types/redux/meters'
+import { selectAllMeters } from '../../redux/api/metersApi';
+import { selectAllGroups, groupsApi } from '../../redux/api/groupsApi';
+
+/**
+ *   Visual graph component that shows the relationship between all groups and meters
+ * entered by an admin
+ * @returns D3 force graph visual
+ */
+interface CreateVisualGroupProps {
+    groups: GroupData[];
+    meters: MeterData[];
+}
+
+//Declare node types for groups and meters, will determine every node's design schema on the graphic
+type GroupNodeType = 'unselectedGroup' | 'selectedGroup' | 'childGroup' | 'deepGroup';
+type MeterNodeType = 'meter' | 'childMeter' | 'deepMeter';
+type AllNodeType = GroupNodeType | MeterNodeType;
+
+export const CreateVisualGroupComponent: React.FC<CreateVisualGroupProps> = ({
+    groups,
+    meters
+}) => {
+
+    const [selectedGroupId, setSelectedGroupId] = useState<number | null>(null);
+
+    const intl = useIntl();
+
+    /*Get Meter data from redux*/
+    const allMeters: MeterData[] = useAppSelector(selectAllMeters);
```

Comment:
I have another comment on the props (did earlier). This seems to create a different copy of meters. Maybe all uses here should be with this one?